ui/src/pages/QuickSetup/AddPartiesPage.tsx

```diff
@@ -170,45 +181,60 @@ const AddPartiesPage = () => {
 
 const CreateVerifiedIdentity = (props: {
   onComplete: () => void;
-  party: PartyDetails;
+  party: PartyToken;
   operator: string;
 }) => {
   const { onComplete, party, operator } = props;
+
+  const completedIdentityVerifications = useIdentityVerification(party);
+  const completedPartyOnboarding = usePartyOnboarding(party, operator);
+
+  useEffect(() => {
+    if (completedIdentityVerifications && completedPartyOnboarding) {
+      return onComplete();
+    }
+  }, [completedIdentityVerifications, completedPartyOnboarding, onComplete]);
+
+  return null;
+};
```

Comment:
Before there was one big useEffect doing multiple things - I split up the jobs into smaller hooks for easier management. Each hook eventually reports the completion status of that job